firmware/defmt-rtt/src/lib.rs

```diff
@@ -47,6 +47,18 @@ unsafe impl defmt::Logger for Logger {
         unsafe { ENCODER.start_frame(do_write) }
     }
 
+    fn flush() {
+        Self::acquire();
```

Comment:
I think you want the `acquire` and `release` one level up in `defmt::flush`. Otherwise every implementation would need to repeat the acquire/release